malebranche/client/instance.py

```diff
@@ -35,8 +32,7 @@ def start_span(level=logging.DEBUG):
         context: ContextManager = ContextManager()
         token = _EXECUTION_LOG_CONTEXT.set(context)
     try:
-        logger = Logger(__name__, context, level)
-        yield Span(logger)
+        yield Span().__enter__()
```

Comment:
If you want to perform some kind of initialization before returning the `Span()` instance, I recommend you to add an `initialize(...)` or similar method (to be called by the `__enter__` magic function internally). The reason to do that is to avoid unbalanced `__enter__`, `__exit__`  calls, that may produce unexpected situations in some cases.